core/src/main/java/org/fenixedu/messaging/domain/MessageDispatchReport.java

```diff
@@ -15,7 +15,18 @@ public Message getMessage() {
     }
 
     public boolean isFinished() {
-        return getTotalCount() == getFailedCount() + getInvalidCount() + getDeliveredCount();
+        return getTotalCount() == getResolvedCount();
+    }
+
+    public Integer getResolvedCount() {
```

Comment:
Yes, but i didn't make or touch that class in the dml. Should I? I'm all for it.